provider/openstack/provider.go

```diff
@@ -1039,14 +1040,62 @@ func (e *Environ) StartInstance(args environs.StartInstanceParams) (*environs.St
 		args.InstanceConfig.MachineId,
 	)
 
+	waitForActiveServerDetails := func(
+		client *nova.Client,
+		id string,
+		timeout time.Duration,
+	) (server *nova.ServerDetail, err error) {
+
+		err = retry.Call(retry.CallArgs{
+			Clock: e.clock,
+			Delay: 10 * time.Second,
+			Func: func() error {
+				server, err = client.GetServer(id)
+				if err == nil {
+					if server.Status == nova.StatusBuild {
+						return errors.New("server has status BUILD, waiting 10 seconds before polling again...")
+					}
+				}
+				return nil
```

Comment:
We should probably have tests for this. So that you don't have to wait for real time to pass, there's a SetClock method on the *openstack.Environ type: see TestStopInstanceSecurityGroupNotDeleted.